### PR TITLE
Added the :fenced_code Hoedown option.

### PR DESCRIPTION
**Pull request by milmazz**
Changes proposed by @dignifiedquire in his `design` branch.

### CODE REVIEW

**Comment by ericmj** [comment]
Looks like the execution permission bit is being set for the changed files.

**Comment by milmazz** [comment]
@ericmj Fixed! Thanks for noticing this mistake, definitively I need to review my `smb.conf`, but that's another issue :)

**Comment by josevalim** [comment]
Honestly, I don't want to support fenced code because it is not part of the markdown "spec" (afaik). Unless we are sure all markdown processors (at least the three we support) handle it, I would avoid adding specific behaviour.

**Comment by ericmj** [comment]
All three markdown engines support fenced code blocks. @milmazz Can you make sure that it's enabled for all of them?

**Comment by josevalim** [comment]
Thanks @ericmj.

Then it is also worth debating if we want to starting allowing fenced code blocks in Elixir documentation. I personally find them a hack and unreadable. :(

**Comment by ericmj** [comment]
I think it's worth the hack since it seems to be the only way to highlight other languages. We don't even have a way to disable the elixir highlighting right now.

**Comment by whatyouhide** [comment]
Big :+1: for fenced code blocks, I actually find them extremely more readable than 4-spaces indentation. Also, I'd love to be able to specify the language to highlight; as of now, gettext documentation is full of PO file snippets highlighted as Elixir :(.

**Comment by milmazz** [comment]
@ericmj Perfect, I'll include the fenced code blocks option for the others Markdown parser.

**Comment by Dignifiedquire** [comment]
Adding my vote to allow fenced code blocks. From my personal preference I think they are much more readable, especially when you have indented code inside the code block itself and from an objective standpoint as they are the only way to specify the language.

**Comment by josevalim** [comment]
Then we also need to support fenced blocks on IO.ANSI.Docs (which is what elixir uses for formatting markdown inside iex).

**Comment by milmazz** [comment]
I already added the `fenced_code_blocks` extension for Pandoc. In the case of the Earmark parser this feature is already included through the option [gfm](https://github.com/elixir-lang/ex_doc/blob/master/lib/ex_doc/markdown/earmark.ex#L28)

**Comment by milmazz** [comment]
According to [this page](http://pandoc.org/releases.html#pandoc-1.10-2013-01-19) `pandoc` supports enable or disable specific syntax extensions by appending them (with + or -) to the writer or reader name (e.g. `markdown+fenced_code_blocks`) from its version 1.10.

I've looked for more information about the [CI environment OS](http://docs.travis-ci.com/user/ci-environment/#CI-environment-OS) provided by Travis and I found that they use _Ubuntu 12.04 LTS Server Edition 64 bit_. Also, I found that the version of the pandoc package for [Ubuntu 12.04](http://packages.ubuntu.com/precise/pandoc) is 1.9.1.1-1. So, that's why this feature does not work in this environment.

The option that I prefer is proceed to modify the `before_install` section in the `.travis.yml` file and then try to include a newer version of the `pandoc` package. But, I really want to know your opinion on this matter first.

P.S. I've run the tests on Debian 8.0 and everything run smoothly.

**Comment by ericmj** [comment]
Installing a newer pandoc manually does sound like the best solution.

File: .travis.yml [hunk 1]
```diff
@@ -5,7 +5,8 @@ notifications:
 otp_release:
   - 17.0
 before_install:
-  - sudo apt-get install -y pandoc
+  - wget -c https://github.com/jgm/pandoc/releases/download/1.13.2/pandoc-1.13.2-1-amd64.deb
+  - sudo dpkg -i pandoc-1.13.2-1-amd64.deb
```
Comment (ericmj) [inline .travis.yml]: :+1:

**Comment by ericmj** [comment]
@josevalim Are we good to ship this now?
@milmazz Thanks! Please rebase into single commit (commits should be in the present tense and should not end with punctuation).

**Comment by josevalim** [comment]
:+1:

**Comment by Dignifiedquire** [comment]
:tada: :balloon:

**Comment by milmazz** [comment]
@ericmj Please let me know is the latest commit fulfill your suggestions.

@josevalim @ericmj @Dignifiedquire @whatyouhide Thanks for your support!

**Comment by ericmj** [comment]
Perfect! Will merge after CI.

**Pull request merged**